src/zkit/CircomZKitManager.ts

```diff
@@ -0,0 +1,144 @@
+import * as fs from "fs";
+import path from "path";
+import process from "process";
+
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { CircomZKit, CircuitZKit, CircuitInfo, ManagerZKitConfig, CompileOptions } from "@solarity/zkit";
+
+import { FileFilterSettings, ZKitConfig } from "../types/zkit-config";
+import { NonExistentCircuitError, CircuitWithoutMainComponentError, UncompiledCircuitError } from "./errors";
+import { MAIN_COMPONENT_REG_EXP } from "../constants";
+
+export class CircomZKitManager {
+  private _circomZKit: CircomZKit;
+
+  constructor(
+    private _hre: HardhatRuntimeEnvironment,
+    private _config: ZKitConfig = _hre.config.zkit,
+  ) {
+    this._circomZKit = new CircomZKit(this._buildManagerZKitConfig());
+
+    this._config.compilationSettings = {
+      ...this._config.compilationSettings,
+      ...this._normalizeFilterSettings(this._config.compilationSettings),
+    };
+    this._config.verifiersSettings = {
+      ...this._config.verifiersSettings,
+      ...this._normalizeFilterSettings(this._config.verifiersSettings),
+    };
+  }
+
+  public async compile() {
+    const circuitsInfo: CircuitInfo[] = this._filterCircuits(this.getCircuitsInfo(), this._config.compilationSettings);
+    const circuits: CircuitZKit[] = [];
+
+    circuitsInfo.forEach((info: CircuitInfo) => {
+      if (info.id !== null && this._hasMainComponent(info)) {
+        circuits.push(this._circomZKit.getCircuit(info.id));
+      }
+    });
+
+    for (let i = 0; i < circuits.length; i++) {
+      await circuits[i].compile(this._buildCompilerOptions());
+    }
+  }
+
+  public async generateVerifiers() {
+    const circuitsInfo: CircuitInfo[] = this._filterCircuits(this.getCircuitsInfo(), this._config.verifiersSettings);
```

Comment:
Let's filter by compilation settings here, too. To avoid confusion when schemas are compiled but verifier generation fails